### PR TITLE
Implement a compiler plugin to validate service semantics of Salesforce listener

### PR DESCRIPTION
**Pull request by sachinira**
## Purpose
- Related to https://github.com/wso2-enterprise/choreo/issues/7328
- Related to https://github.com/ballerina-platform/module-ballerinax-sfdc/issues/160

## Automation tests
 - Unit tests 
   > Done for 8 diagnostic scenarios
 - Integration tests
   > Not done

Ballerina SL Beta 2
Ubuntu 20.04 LTS

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,4 +1,21 @@
 org.gradle.caching=true
 group=org.ballerinalang.sf
 version=2.2.1
-ballerinaLangVersion=2.0.0-beta.2.1
+ballerinaLangVersion=2.0.0-beta.2
```
Comment (LakshanSS) [inline gradle.properties:4]: `2.0.0-beta.2.1` is the latest and correct version for ballerinaLangVersion.
Don't change this to `2.0.0-beta.2`.

But for the other stdlib versions, u can use beta.2
Ex: `stdlibIoVersion=0.6.0-beta.2`

File: gradle.properties [hunk 2]
```diff
@@ -1,4 +1,21 @@
...
+stdlibFileVersion=0.7.0-beta.2
+stdlibCryptoVersion=1.1.0-beta.2
+stdlibLogVersion=1.1.0-beta.2
+stdlibMimeVersion=1.1.0-beta.2
+stdlibCacheVersion=2.1.0-beta.2
+stdlibAuthVersion=1.1.0-beta.2
+stdlibJwtVersion=1.1.0-beta.2
+stdlibOAuth2Version=1.1.0-beta.2
+stdlibUuidVersion=0.10.0-beta.2
+stdlibHttpVersion=1.1.0-beta.2
```
Comment (LakshanSS) [inline gradle.properties:21]: Better to add these in ascending order.
So it be more readable.

File: build.gradle [hunk 3]
```diff
@@ -59,6 +63,27 @@ task createBalaDirectory {
...
+//        }
+//    }
+//    copy {
+//        from file("${project.projectDir}/sfdc/target/cache")
+//        exclude '**/*-testable.jar'
+//        exclude '**/tests_cache/'
+//        into file("${artifactCacheParent}/cache/")
+//    }
+//    outputs.dir artifactCacheParent
+//}
```
Comment (LakshanSS) [inline build.gradle:85]: Why is it commented?
Do we need this task?

File: gradle.properties (hunk 1 above)
Comment (sachinira) [reply gradle.properties:4]: I have used a dependency called jballerina-tools to build the ballerina project during the gradle build. When I try to import it in 2.0.0-beta2.1 version it is not available. So as to be consistent with the ballerina versions I have used this lang version. Have to check if some version related problems occur when using different versions of ballerina lang and jballerina tools.

File: compiler-plugin/src/main/java/org/ballerinalang/sf/plugin/Constants.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (sachinira) [reply compiler-plugin/src/main/java/org/ballerinalang/sf/plugin/Constants.java]: Fixed

File: gradle.properties (hunk 2 above)
Comment (sachinira) [reply gradle.properties]: Sorted

File: build.gradle (hunk 3 above)
Comment (sachinira) [reply build.gradle]: Removed this one it was unnecessary

**Comment by sachinira** [comment]
Closing the PR due to conflicting commit history

**Pull request closed without merging**